### PR TITLE
test(e2e): retire the last three CE skips — and fix the two product bugs they exposed

### PR DESCRIPTION
**Pull request by jmrplens**
## Summary

Following up on the skip-free push: the last three CE skips are converted into real coverage, and — proving the point of the exercise — the new coverage immediately exposed **two real product bugs**, both fixed here.

### The three skips

1. **`TestIndividual_CICatalogGet`** — now publishes a real CI/CD catalog version through the Docker runner: component + README + release job committed, a semver tag triggers the pipeline, and `release-cli` is invoked explicitly with the compose-network server URL (GitLab pins the predefined CI server variables against yaml overrides, and the instance's `external_url` resolves to the job container itself). The final GET polls briefly because catalog publication is asynchronous after the release lands.
2. **`TestIndividual_UserApproveReject`** — consumes two users `setup-gitlab.sh` now seeds in `blocked_pending_approval` state (created via the API, state flipped through the Rails console — exactly what a real pending signup produces, which the API alone cannot). Approve re-activates one; reject deletes the other; the instance ends clean.
3. **`TestMeta_GroupBoardListColumns`** — moved behind the enterprise gate (`groupboardextras_ee_test.go`): provisioning a group issue board via API is a Premium capability (CE 404s on the create route). On EE, provisioning is *asserted*, not skipped, and the reorder subtest creates a second label list because GitLab rejects moving a board's only list.

### The two product bugs the new coverage caught

- **`gitlab_get_catalog_resource` / `gitlab_list_catalog_resources` were broken against current GitLab**: their GraphQL queries used `CiCatalogResource` fields GitLab has removed (`webUrl`, `forksCount`, `openIssuesCount`, `openMergeRequestsCount`, `readmeHtml`, `latestVersion`), so every live call failed validation and was misreported as "not found". The tools had only ever been exercised against httptest mocks. Queries and output shapes now mirror the current schema 1:1 (`web_path`, `last_30_day_usage_count`, `archived`, `topics`, `verification_level`, `visibility_level`; README and components sourced from the newest version; `semver` flattened from its major/minor/patch object; components read through their connection). Docs, markdown formatters, and golden tool snapshots updated.
- **`gitlab_group_board_list_update` could never decode a successful response**: client-go's group-level `UpdateIssueBoardList` declares `[]*BoardList`, but GitLab returns the single updated object. The handler issues the request directly until the upstream fix lands — filed as [gitlab-org/api/client-go!2996](https://gitlab.com/gitlab-org/api/client-go/-/merge_requests/2996) following their contribution process (`make reviewable`, mock regeneration, test fixture corrected). The unit mock that had reproduced the wrong array shape is fixed too.

## Validation

- Full CE docker run: **1453 tests, 0 failures, 0 skips** — the suite's first completely skip-free run.
- Targeted EE run: board list CRUD green end to end (create, get, reorder, delete).
- Full unit suite, `golangci-lint`, `check-test-goroutines`, doc-coverage, catalog-manifest, llms, footprint, and stats gates all pass.

<a href="https://cubic.dev/pr/jmrplens/gitlab-mcp-server/pull/282?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

<div id='description'>
<a href="https://bito.ai#summarystart"></a><h3>Summary by Bito</h3><ul><li>Updated CI/CD Catalog resource models to include new fields like usage count, topics, and verification level.</li>
<li>Refactored GraphQL queries for CI/CD Catalog resources to fetch updated fields and simplify version information.</li>
<li>Updated documentation and test data to reflect changes in the CI/CD Catalog schema.</li>
<li>Added new test cases and updated existing E2E tests to accommodate the schema changes.</li>
</ul></div>

## Summary by Sourcery

Retire the remaining CE end-to-end skips while updating GitLab integrations for current catalog and group board APIs.

New Features:
- Align CI/CD Catalog resource retrieval with the current GitLab GraphQL schema and expose updated resource, version, and component metadata.
- Add skip-free end-to-end coverage for catalog publication, user approval/rejection, and enterprise group board-list CRUD.

Bug Fixes:
- Fix CI/CD Catalog tools reporting live resources as missing because they queried removed GitLab GraphQL fields.
- Fix group board-list updates by decoding GitLab’s single-object response instead of the incompatible client library array response.

Enhancements:
- Move group board-list coverage to the enterprise test suite and make successful board provisioning mandatory there.
- Seed pending-approval users and publish catalog versions through real CI pipelines for reliable end-to-end coverage.
- Improve statistics scanning so Go code embedded in raw-string fixtures is not counted.

Documentation:
- Update CI/CD Catalog reference documentation, markdown output expectations, testing statistics, and generated project metrics.

Tests:
- Expand CI/CD Catalog schema and edge-case unit coverage, correct group board-list response fixtures, and cover raw request construction failures.

## Summary by CodeRabbit

* **New Features**
  * CI Catalog resources now display web paths, usage, topics, verification, visibility, archive status, and release/version details.
  * Added enterprise end-to-end coverage for group board list management.
  * Added catalog publishing coverage using real CI pipelines.

* **Updates**
  * Refreshed tool response schemas and documentation to reflect the latest catalog fields.
  * Updated testing, coverage, and token-footprint statistics.
  * Improved pending-user approval and rejection test setup.

* **Bug Fixes**
  * Corrected group board list updates to use the current single-object API response format.

### CODE REVIEW

File: internal/tools/cicatalog/cicatalog_test.go [hunk 1]
```diff
@@ -21,36 +21,54 @@ const sampleResourceNode = `{
...
+			"name": "2.0.0",
+			"releasedAt": "2026-03-01T08:00:00Z",
+			"createdAt": "2026-03-01T07:59:00Z",
+			"semver": {"major": 2, "minor": 0, "patch": 0},
+			"path": "/my-group/go-pipeline/-/tags/2.0.0",
+			"readmeHtml": null,
+			"components": {"nodes": [
+				{"name": "build", "description": null, "includePath": "gitlab.example.com/my-group/go-pipeline/build@2.0.0", "inputs": []}
+			]}
+		}
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply internal/tools/cicatalog/cicatalog_test.go:70]: Applied in cdf34857 — TestGet_ByFullPath now asserts every schema-mirroring field (semver, created_at, path, archived, topics, visibility_level, verification_level, last_30_day_usage_count) via an assertSampleDetailFields helper, plus two new shape tests (TestGet_NoVersions, TestGet_PartialSemverStaysEmpty). Package coverage is 100.0%.

File: internal/tools/cicatalog/cicatalog.go [hunk 2]
```diff
@@ -234,24 +261,47 @@ func nodeToResourceDetail(n gqlResourceNode) ResourceDetail {
...
+	}
+	if v.Components != nil {
+		item.Components = convertComponents(v.Components.Nodes)
+	}
+	if v.CreatedAt != nil {
+		item.CreatedAt = *v.CreatedAt
+	}
+	if v.Semver != nil {
+		item.Semver = fmt.Sprintf("%d.%d.%d", v.Semver.Major, v.Semver.Minor, v.Semver.Patch)
+	}
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply internal/tools/cicatalog/cicatalog.go]: Applied in cdf34857 — gqlSemver decodes major/minor/patch as *int and item.Semver is set only when all three are present, so a partial semver stays empty instead of collapsing into '0.0.0'. TestGet_PartialSemverStaysEmpty covers it. No prerelease field added, as you note the schema does not expose one.

File: internal/tools/cicatalog/markdown.go [hunk 3]
```diff
@@ -59,11 +58,18 @@ func writeCatalogResourceSummary(sb *strings.Builder, r ResourceDetail) {
 	sb.WriteString("| Field | Value |\n|-------|-------|\n")
 	fmt.Fprintf(sb, "| ID | %s |\n", toolutil.EscapeMdTableCell(r.ID))
 	fmt.Fprintf(sb, "| Full Path | %s |\n", toolutil.EscapeMdTableCell(r.FullPath))
-	fmt.Fprintf(sb, "| URL | [%s](%s) |\n", toolutil.EscapeMdTableCell(r.FullPath), r.WebURL)
+	fmt.Fprintf(sb, "| Web Path | %s |\n", toolutil.EscapeMdTableCell(r.WebPath))
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply internal/tools/cicatalog/markdown.go]: Applied in cdf34857 — the detail summary renders Web Path as an escaped Markdown link, matching the list formatter, with an output assertion.

File: test/e2e/scripts/setup-gitlab.sh [hunk 4]
```diff
@@ -643,7 +643,53 @@ if command -v docker >/dev/null 2>&1; then
...
+            echo "      WARNING: could not create pending-${role} user; approve/reject happy path will skip"
+            pending_ids=""
+            break
+        fi
+        pending_ids="${pending_ids} ${pending_id}"
+    done
+    if [ -n "${pending_ids}" ]; then
+        pending_ids_csv=$(echo ${pending_ids} | tr ' ' ',')
+        if docker compose -f "${COMPOSE_FILE}" exec -T gitlab gitlab-rails runner \
+            "User.where(id: [${pending_ids_csv}]).update_all(state: 'blocked_pending_approval')" \
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply test/e2e/scripts/setup-gitlab.sh:687]: Applied in cdf34857 — each seeded user ID is digits-only-validated (case pattern) before being interpolated into the Rails expression; anything else empties the ID and the seeding warn-skips.

File: test/e2e/suite/groupboardextras_ee_test.go [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+			"params": map[string]any{
+				"group_id": grp.gidStr(),
+				"board_id": boardID,
+				"list_id":  secondListID,
+				"position": 0,
+			},
+		})
+		requireNoError(t, err, "group_board_update_list")
+		requireTruef(t, out.ID > 0, "expected a board list in the update response")
+		t.Logf("Reordered board list %d to position %d", out.ID, out.Position)
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply test/e2e/suite/groupboardextras_ee_test.go:111]: Applied in cdf34857 — the reorder subtest asserts the response is the moved list (ID == secondListID) at position 0.

File: llms-full.txt [hunk 6]
```diff
@@ -1818,15 +1818,15 @@ Annotations: readOnly=true, destructive=false, idempotent=true, openWorld=true
...
+{"additionalProperties":false,"properties":{"next_steps":{"items":{"type":"string"},"type":["null","array"]},"pagination":{"additionalProperties":false,"properties":{"end_cursor":{"type":"string"},"has_next_page":{"type":"boolean"},"has_previous_page":{"type":"boolean"},"start_cursor":{"type":"string"}},"required":["has_next_page","has_previous_page"],"type":"object"},"resources":{"items":{"additionalProperties":false,"properties":{"archived":{"type":"boolean"},"description":{"type":"string"},"full_path":{"type":"string"},"icon":{"type":"string"},"id":{"type":"string"},"last_30_day_usage_count":{"type":"integer"},"latest_released_at":{"type":"string"},"latest_version_name":{"type":"string"},"name":{"type":"string"},"star_count":{"type":"integer"},"topics":{"items":{"type":"string"},"type":["null","array"]},"verification_level":{"type":"string"},"visibility_level":{"type":"string"},"web_path":{"type":"string"}},"required":["id","name","full_path","web_path","star_count","last_30_day_usage_count","archived"],"type":"object"},"type":["null","array"]}},"required":["resources","pagination"],"type":"object"}
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply llms-full.txt]: Applied in cdf34857 — gqlVersion.ReleasedAt decodes as *string, copied only when non-null, and released_at carries omitempty so the schema no longer requires it. Golden snapshots and generated surfaces (llms, footprint) regenerated.

File: README.md [hunk 7]
```diff
@@ -415,19 +415,19 @@ and is never logged. Full details: [PRIVACY.md](PRIVACY.md).
...
 
 | Category                        |  Count |
 | ------------------------------- | -----: |
-| Source functions                |  7,511 |
+| Source functions                |  7,512 |
 | — exported (public)             |  2,633 |
-| — unexported (private)          |  4,878 |
-| Unit test functions (`TestXxx`) | 11,686 |
+| — unexported (private)          |  4,879 |
+| Unit test functions (`TestXxx`) | 11,684 |
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply README.md]: Applied in cdf34857 — scanGoFile now tracks multi-line raw-string state and skips those lines for every counter, so fixture sources embedding fake 'func Test...' lines (the goroutine-audit fixtures) no longer inflate the counts. Chosen over per-file AST parsing because it fixes all line counters (subtests, err-checks, nolints) in one place at zero parse cost; a unit test covers the fixture case. README/testing.md regenerated with the corrected numbers.

File: internal/tools/testdata/tools_individual.json [hunk 8]
```diff
@@ -31953,6 +31953,9 @@
         "resource": {
           "additionalProperties": false,
           "properties": {
+            "archived": {
+              "type": "boolean"
+            },
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply internal/tools/testdata/tools_individual.json:31958]: Partially applied in cdf34857 — released_at and web_path (the fields that are genuinely null in practice: draft resources and partial versions) now carry omitempty and are no longer schema-required, and semver handles its nullable components. name/full_path stay required deliberately: a catalog resource or version without its identity is not meaningfully representable, the codebase convention models optional output strings as omitempty rather than pointers, and loosening identity fields would ripple through every consumer for a null GitLab does not emit in practice. Golden snapshots regenerated with the existing snapshot test as suggested.
